enhancements/insights/on-demand-data-gathering.md

```diff
@@ -0,0 +1,255 @@
+
+
+### Risks and Mitigations
+
+Risk: Inconsistent Insights analysis results. Assume following situation:
+- periodic gathering is running in the Insights operator (let's call it gather A)
+- user runs a new on demand data gathering (let's call it gather B) e.g 30 seconds after gather A
+- gather A and B finish at essentially the same time
+- the external processing service (in the console.redhat.com) provides only one recent Insights analysis result, and there is no guarantee which (of the two) it is.
+- this can be very confusing, because user running gather B can get result of the gather A (e.g there could be different gatherers enabled, which would produce different results)
+
+Mitigation: This is the very first question in the [Open questions](#open-questions) section, but perhaps reporting the Insights analysis result in one place could slightly mitigate this problem (the status part could tell whether it was a periodic gather or user-triggered gather). 
+Another option would be to disallow the parallel gather execution - if the Insights operator sees a new CR instance (or any other running gather process) then it interrupts any gathering in progress. 
+
+### Drawbacks
+
+- Very frequent data gathering in a cluster can produce a significant load on the processing part in the console.redhat.com and can overwhelm the processing pipeline. 
+Theoretically, we could limit the number of user-trigerred gathering over time, but this problem can happen today when something is deleting the Insights operator pod on a regular basis with appropriate timing (so that data can be uploaded).  
+- When the newly spawned container is failing to upload the data then it can be running for some longer time (couple of hours). 
+
+
+## Design Details
+
+### Open Questions
+- How to solve the problem with only one Insights analysis result available in a cluster? How to avoid the "uncertainty" of the Insights analysis results described in the [Risk and Mitigations](#risks-and-mitigations) section? 
+- Should the newly created gathering process update the clusteroperator conditions (e.g it fails to upload the data, should it update the `Degraded` status as it is today)? This can probably be racy too or it can lead to temporary and inconsistent conditions state. 
+- What all is required to spawn an extra container? Do we need a new image and a new binary (note that the Insights operator currently includes some additional features - such as pulling SCA certificates - that are not required for a new container/process)?
+- Is some clean-up of the old `datagather.insights.openshift.io` CRs (and completed containers/pods) required? I think it would make sense to keep them only for 24h.
```

Comment:
Garbage collecting them makes sense to me, perhaps this could default to 24 hours (as with CSRs) but then be configurable via the operator configuration

Reply:
Yes. Agree.

Reply:
Updated.